docs/src/introduction.md

```diff
@@ -0,0 +1,184 @@
+- `set` is used to change the value of an existing variable
+  - This will error if you try to set a variable that doesn't exist
+
+### Lists
+
+Stack has a built-in list type that can be used to store multiple values. Lists are created with the `()` syntax. When pushed to the stack, the data inside lists **will be evaluated**. The evaluation happens from left to right, and the result is reimplemented into the list. For example, the list below adds two numbers and is collapsed down into the result.
+
+```clojure
+;; Pushes a list to the stack and evaluates it () -> ((4))
+(2 2 +)
+```
+
+However, you can also reference variables in lists and they will be evaluated when the list is pushed to the stack.
+
+```clojure
+;; `my-var` is defined as 0
+0 'my-var def
+
+;; Pushes a list to the stack and evaluates it () -> ((0))
+(my-var)
+```
+
+The symbol `my-var` in the list will be evaluated to its value, 0, when the list is pushed to the stack.
+
+#### Laziness
+
+Just like [variables](#variables), lists can be made "lazy" by using a `'` -> `'(2 2 +)`. This will prevent the list from being evaluated when it is pushed to the stack.
+
+```clojure
+;; Pushes a list to the stack and doesn't evaluate it () -> ((2 2 +))
+'(2 2 +)
+```
+
+In this case, all items in the list are kept as they are.
+
+Alternatively, you can make only certain items in a list lazy `'`, like so:
+
+```clojure
+;; `my-var` is defined as 0
+0 'my-var def
+
+;; Pushes a list to the stack and doesn't evaluate the first item () -> ((my-var 0))
+('my-var my-var)
+```
+
+In this case, the first item in the list is kept as a lazy symbol, while the second item is evaluated.
+
+### Functions
+
+Functions are the primary way to define new operations in Stack. Uniquely, Stack doesn't have a special data type for functions. Instead, functions are just lists of operations. Heres what they look like:
+
+```clojure
+;; Push a function (a lazy list) to the stack () -> ('(...))
+'(fn 1 +)
+
+;; Push a lazy symbol to name the function ('(...)) -> ('(...) 'add-one)
+'add-one
+
+;; Push and evaluate the `def` symbol ('(...) 'add-one) -> ()
+def
+
+;; Push 0 to the stack () -> (0)
+0
+
+;; Push `add-one` to the stack and evaluate it (0) -> (1)
+add-one
+```
+
+The `fn` symbol at the beginning of the list tells the evaluator to treat the list as a function. If this was ommitted, each instruction on the list would be evaluated separately.
+
+```clojure
+;; Push a lazy list to the stack () -> ('(1 +))
+'(1 +)
+
+;; Push a lazy symbol to name the function ('(1 +)) -> ('(1 +) 'add-one)
+'add-one
+
+;; Push and evaluate the `def` symbol ('(1 +) 'add-one) -> ()
+def
+
+;; Push 0 to the stack () -> (0)
+0
+
+;; Push the value of `add-one` to the stack (0) -> (0 (1 +))
+add-one
+
+;; Evaluate the code in the list (0 (1 +)) -> (0)
+call
+```
+
+This is known as *auto-call*, where `fn` at the beginning of the list tells the evaluator to push `call` to the stack after the list, which will evaluate the list.
```

Comment:
This is a little strange. It mentions that there are no functions and then seemingly contradicts itself by saying that there are functions. A clearer distinction would help mitigate this.